File: src/main/java/com/itsci/mju/maebanjumpen/entity/Transaction.java
package com.itsci.mju.maebanjumpen.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "transactions")
@AllArgsConstructor
@NoArgsConstructor
@Data
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer transactionId;

    /** ประเภทของธุรกรรม เช่น Deposit, Withdraw */
    @Column(name = "transaction_type", nullable = false, length = 255)
    private String transactionType;

    /** จำนวนเงิน */
    @Column(nullable = false)
    private Double transactionAmount;

    /** วันที่ทำธุรกรรม */
    @Column(name = "transaction_date", nullable = false)
    private LocalDateTime transactionDate;

    /** สถานะธุรกรรม เช่น PENDING, APPROVED, REJECTED */
    @Column(name = "transaction_status", nullable = false, length = 255)
    private String transactionStatus;

    /** ผู้ทำธุรกรรม (Member: Hirer หรือ Housekeeper) */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    @ToString.Exclude
    private Member member;

    /** เบอร์พร้อมเพย์ (กรณีถอนเงิน) */
    @Column(name = "prompay_number", length = 50)
    private String prompayNumber;

    /** เลขบัญชีธนาคาร (กรณีถอนเงิน) */
    @Column(name = "bank_account_number", length = 50)
    private String bankAccountNumber;

    /** ชื่อบัญชีธนาคาร */
    @Column(name = "bank_account_name", length = 255)
    private String bankAccountName;

    /** วันที่อนุมัติธุรกรรม */
    @Column(name = "transaction_approval_date")
    private LocalDateTime transactionApprovalDate;
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/entity/Transaction.kt
package com.itsci.mju.maebanjumpen.entity

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "transactions")
@JsonIgnoreProperties("hibernateLazyInitializer", "handler")
data class Transaction(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null,

    @Column(name = "transaction_type", nullable = false, length = 255)
    var transactionType: String = "",

    @Column(nullable = false)
    var transactionAmount: Double = 0.0,

    @Column(name = "transaction_date", nullable = false)
    var transactionDate: LocalDateTime? = null,

    @Column(name = "transaction_status", nullable = false, length = 255)
    var transactionStatus: String = "",

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    var member: Member? = null,

    @Column(name = "prompay_number", length = 50)
    var prompayNumber: String? = null,

    @Column(name = "bank_account_number", length = 50)
    var bankAccountNumber: String? = null,

    @Column(name = "bank_account_name", length = 255)
    var bankAccountName: String? = null,

    @Column(name = "transaction_approval_date")
    var transactionApprovalDate: LocalDateTime? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Transaction) return false
        return id != null && id == other.id
    }

    override fun hashCode(): Int = id?.hashCode() ?: 0

    override fun toString(): String = "Transaction(transactionId=$id, transactionType='$transactionType', transactionAmount=$transactionAmount)"
}

